homeassistant/components/asuswrt/sensor.py

```diff
@@ -54,10 +60,41 @@ def state(self):
 
     async def async_update(self):
         """Fetch status from asuswrt."""
+        self._devices = await self._api.async_get_connected_devices()
         self._rates = await self._api.async_get_bytes_total()
         self._speed = await self._api.async_get_current_transfer_rates()
 
 
+class AsuswrtDevicesSensor(AsuswrtSensor):
+    """Representation of a asuswrt download speed sensor."""
+
+    _name = "Asuswrt Devices Connected"
+
+    @property
+    def device_state_attributes(self):
+        """Return the attributes of the sensor."""
+        return self._attributes
+
+    async def async_update(self):
+        """Fetch new state data for the sensor."""
+        await super().async_update()
+        if self._devices:
+            self._state = len(self._devices)
+            mac_addresses = []
+            ip_addresses = []
+            hosts = []
+            for mac in self._devices:
+                device = self._devices[mac]._asdict()
+                mac_addresses.append(device["mac"])
+                ip_addresses.append(device["ip"])
+                hosts.append(device["name"])
+            self._attributes = {
+                "mac_addresses": mac_addresses,
+                "ip_addresses": ip_addresses,
+                "hosts": hosts,
```

Comment:
We try to avoid grouping values as attributes. Can we instead make one binary sensor per connected device with device class `connectivity`?

Reply:
Well we have device_tracker for that so I guess I'll remove the attributes